dora/core/common/src/main/java/alluxio/membership/WorkerClusterView.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.membership;
+
+import alluxio.collections.IndexDefinition;
+import alluxio.collections.IndexedSet;
+import alluxio.wire.WorkerIdentity;
+import alluxio.wire.WorkerInfo;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.Iterators;
+
+import java.time.Instant;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Stream;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * Snapshot for a cluster view.
+ */
+@Immutable
+public final class WorkerClusterView implements Iterable<WorkerInfo> {
+  private final Instant mInstantCreated;
+  private final IndexedSet<WorkerInfo> mWorkers;
+
+  private static final IndexDefinition<WorkerInfo, WorkerIdentity> INDEX_WORKER_ID =
+      IndexDefinition.ofUnique(WorkerInfo::getIdentity);
```

Comment:
there's only one index so i still prefer the original easier approach i proposed: 1 HashSet of live workers and 1 HashSet for lost workers, both indexed by WorkerIdentity

Reply:
changed to use an ordinary map. Can't have two sets of live workers and failed workers as what's contained in the view is determined by the membership manager.